src/components/PDFThumbnail/index.tsx

```diff
@@ -0,0 +1,48 @@
+// @ts-expect-error - This line imports a module from 'pdfjs-dist' package which lacks TypeScript typings.
+import pdfWorkerSource from 'pdfjs-dist/legacy/build/pdf.worker';
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+import {Document, pdfjs, Thumbnail} from 'react-pdf';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import useThemeStyles from '@hooks/useThemeStyles';
+import addEncryptedAuthTokenToURL from '@libs/addEncryptedAuthTokenToURL';
+import type PDFThumbnailProps from './types';
+
+if (!pdfjs.GlobalWorkerOptions.workerSrc) {
+    pdfjs.GlobalWorkerOptions.workerSrc = URL.createObjectURL(new Blob([pdfWorkerSource], {type: 'text/javascript'}));
+}
+
+function PDFThumbnail({previewSourceURL, style, isAuthTokenRequired = false, enabled = true, onPassword = () => {}}: PDFThumbnailProps) {
+    const styles = useThemeStyles();
+
+    const thumbnail = useMemo(
+        () => (
+            <Document
+                loading={<FullScreenLoadingIndicator />}
```

Comment:
Coming for this issue https://github.com/Expensify/App/issues/39356 , since we are not passing the error prop to the Document, the Document will display the message 'Failed to load PDF file.' text in case of an error. This, however, causes a UI issue in the dark theme of our app. we should have passed a custom error component here to avoid UI issue.